### PR TITLE
Use egcc on OpenBSD to fix libxml compilation

### PR DESCRIPTION
**Pull request by frenkel**
First PR for OpenBSD support. #1515 it was suggested that I should also create a PR for the documentation. I'll submit that to the other repo in a few minutes.

### CODE REVIEW

**Comment by parhs** [comment]
This is a good fix. However I precompiled with CC=egcc in openbsd 5.9  but in 6.0 it can't be loaded, it needs to be recompiled

**Comment by frenkel** [comment]
Yes, that's always when library versions change (which happens quite a lot between OpenBSD releases).

**Comment by parhs** [comment]
So is there a solution to have both precompiled ?

**Comment by frenkel** [comment]
No, when you run OpenBSD 6.0 you don't know what 6.1 will contain, so you can only compile it for your current version. Same problem is on Mac OS X when Apple decides to update some important library dependency.

File: ext/nokogiri/extconf.rb [hunk 1]
```diff
@@ -387,7 +391,16 @@ def using_system_libraries?
   do_clean
 end
 
-RbConfig::MAKEFILE_CONFIG['CC'] = ENV['CC'] if ENV['CC']
```
Comment (knu) [inline ext/nokogiri/extconf.rb]: What about leaving this unchanged and put `if openbsd? && !ENV['CC'] ... ENV['CC'] = find_executable 'egcc' end` above it so you could still override `CC`?

Comment (frenkel) [inline ext/nokogiri/extconf.rb]: Good point! I've changed it.

**Comment by knu** [comment]
Out of curiosity, is clang not a choice on OpenBSD?  I heard it started to bundle clang starting from 6.0.

**Comment by frenkel** [comment]
Well, they imported in CVS after the 6.0 release, but I believe they won't make it a supported choice in 6.1. Clang can be installed on amd64 using packages in 5.9 and 6.0 already though.

**Comment by knu** [comment]
Committed, thanks.

**Pull request merged**